### PR TITLE
Merge xcode11 into d16-3.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

File: Make.config [hunk 1]
```diff
@@ -9,6 +9,9 @@ $(TOP)/Make.config.inc: $(TOP)/Make.config
 	@printf "MAC_COMMIT_DISTANCE:=$(shell LANG=C; export LANG && git --git-dir $(TOP)/.git log `git --git-dir $(TOP)/.git blame -- ./Make.versions HEAD | grep MAC_PACKAGE_VERSION= | sed 's/ .*//' `..HEAD --oneline | wc -l | sed 's/ //g')\n" >> $@
 	@if which ccache > /dev/null 2>&1; then printf "ENABLE_CCACHE=1\nexport CCACHE_BASEDIR=$(abspath $(TOP)/..)\n" >> $@; echo "Found ccache on the system, enabling it"; fi
 	@if test -d $(TOP)/../maccore; then printf "ENABLE_XAMARIN=1\n" >> $@; echo "Detected the maccore repository, automatically enabled the Xamarin build"; fi
+	@# Build from source for Xcode 11 support until mono has packages
+	@# TODO: to be removed when we can use the mono archive again.
```
Comment (dalexsoto) [inline Make.config]: Do we still need this? can't we use mono binaries?

File: src/Constants.iOS.cs.in [hunk 2]
```diff
@@ -135,5 +135,13 @@ namespace MonoTouch {
...
 		public const string IdentityLookupUILibrary = "/System/Library/Frameworks/IdentityLookupUI.framework/IdentityLookupUI";
 		public const string NaturalLanguageLibrary = "/System/Library/Frameworks/NaturalLanguage.framework/NaturalLanguage";
+		// iOS 13.0
+		public const string SoundAnalysisLibrary = "/System/Library/Frameworks/SoundAnalysis.framework/SoundAnalysis";
+		public const string BackgroundTasksLibrary = "/System/Library/Frameworks/BackgroundTasks.framework/BackgroundTasks";
+		public const string PencilKitLibrary = "/System/Library/Frameworks/PencilKit.framework/PencilKit";
+		public const string QuickLookThumbnailingLibrary = "/System/Library/Frameworks/QuickLookThumbnailing.framework/QuickLookThumbnailing";
+		public const string LinkPresentationLibrary = "/System/Library/Frameworks/LinkPresentation.framework/LinkPresentation";
+		public const string VisionKitLibrary = "/System/Library/Frameworks/VisionKit.framework/VisionKit";
+		public const string CoreHapticsLibrary = "/System/Library/Frameworks/CoreHaptics.framework/CoreHaptics";
```
Comment (mandel-macaque) [inline src/Constants.iOS.cs.in]: @rolfbjarne @spouliot we said we had to sort this guys out, but there are not sorted at all, should we get the chance to do it now?

File: src/Constants.mac.cs.in [hunk 3]
```diff
@@ -153,5 +153,20 @@ namespace MonoMac {
...
+		public const string ExecutionPolicyLibrary = "/System/Library/Frameworks/ExecutionPolicy.framework/ExecutionPolicy";
+		public const string FileProviderLibrary = "/System/Library/Frameworks/FileProvider.framework/FileProvider";
+		public const string FileProviderUILibrary = "/System/Library/Frameworks/FileProviderUI.framework/FileProviderUI";
+		public const string PushKitLibrary = "/System/Library/Frameworks/PushKit.framework/PushKit";
+		public const string QuickLookThumbnailingLibrary = "/System/Library/Frameworks/QuickLookThumbnailing.framework/QuickLookThumbnailing";
+		public const string SoundAnalysisLibrary = "/System/Library/Frameworks/SoundAnalysis.framework/SoundAnalysis";
+		public const string PencilKitLibrary = "/System/Library/Frameworks/PencilKit.framework/PencilKit";
+		public const string SpeechLibrary = "/System/Library/Frameworks/Speech.framework/Speech";
+		public const string LinkPresentationLibrary = "/System/Library/Frameworks/LinkPresentation.framework/LinkPresentation";
+		public const string CoreHapticsLibrary = "/System/Library/Frameworks/CoreHaptics.framework/CoreHaptics";
```
Comment (mandel-macaque) [inline src/Constants.mac.cs.in]: Same as above.

File: src/Constants.watch.cs.in [hunk 4]
```diff
@@ -60,5 +60,13 @@ namespace ObjCRuntime {
...
 		public const string NaturalLanguageLibrary = "/System/Library/Frameworks/NaturalLanguage.framework/NaturalLanguage";
 		public const string MediaPlayerLibrary = "/System/Library/Frameworks/MediaPlayer.framework/MediaPlayer";
+
+		// WatchOS 6
+		public const string AuthenticationServicesLibrary = "/System/Library/Frameworks/AuthenticationServices.framework/AuthenticationServices";
+		public const string NetworkLibrary = "/System/Library/Frameworks/Network.framework/Network";
+		public const string PushKitLibrary = "/System/Library/Frameworks/PushKit.framework/PushKit";
+		public const string SoundAnalysisLibrary = "/System/Library/Frameworks/SoundAnalysis.framework/SoundAnalysis";
+		public const string CoreMediaLibrary = "/System/Library/Frameworks/CoreMedia.framework/CoreMedia";
+
```
Comment (mandel-macaque) [inline src/Constants.watch.cs.in]: Same as above.

File: docs/website/mtouch-errors.md [hunk 5]
```diff
@@ -2826,6 +2875,12 @@ There are two main reasons for this:
   means (some build options causes the exact list of dynamic symbols to vary).
 
 <!-- 5219 and 5220 used by mmp -->
+=======
```
Comment (VincentDondain) [inline docs/website/mtouch-errors.md]: Conflict leftover

File: mk/xamarin.mk [hunk 6]
```diff
@@ -7,7 +7,7 @@ MONO_BRANCH    := $(shell cd $(MONO_PATH) 2> /dev/null && git symbolic-ref --sho
 endif
 
 ifdef ENABLE_XAMARIN
-NEEDED_MACCORE_VERSION := b7f1347fcf4bd388986c71c1fc1138acf1ffa35d
+NEEDED_MACCORE_VERSION := 7d6d7bd9ed18a2a8e29d798950ca1ee8ccd09b5d
```
Comment (VincentDondain) [inline mk/xamarin.mk]: I don't see xcode11 merged into d16-3 for maccore and this https://github.com/xamarin/maccore/commit/7d6d7bd9ed18a2a8e29d798950ca1ee8ccd09b5d doesn't exist

File: Make.config (hunk 1 above)
Comment (dalexsoto) [reply Make.config]: Answering myself from a conversation with Rolf, we will continue to build our own mono until mono binaries uses xcode11 final to build the binaries.

**Comment by mandel-macaque** [comment]
Added donotmerge until we get the complete commit history.

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-6980/1/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-6980/1/display/redirect) (on internal Jenkins) failed in stage(s) 'Build' 🔥 : hudson.AbortException: script returned exit code 2

✅ [Provisioning succeeded](https://jenkins.internalx.com/job/macios/job/PR-6980/1//console)
🔥 [Build failed](https://jenkins.internalx.com/job/macios/job/PR-6980/1//console) 🔥

File: src/Constants.iOS.cs.in (hunk 2 above)
Comment (rolfbjarne) [reply src/Constants.iOS.cs.in]: We can sort later, I'd prefer to not make this PR more complicated than it already is.

**Pull request closed without merging**